common/components/forms/FormValidation.jsx

```diff
@@ -0,0 +1,54 @@
+// @flow
+
+import React from 'react';
+import _ from 'lodash'
+
+export type Validation = {|
+  checkFunc: (formState: any) => boolean,
+  errorMessage: string
+|};
+
+type Props = {|
+  validations: $ReadOnlyArray<Validation>,
+  onValidationCheck: (boolean) => void,
+  formState: any
+|};
+type State = {|
+  errorMessages: Array<string>
+|};
+
+/**
+ * Performs validations for forms, displaying errors in the markup where the component is placed
+ */
+class FormValidation extends React.PureComponent<Props,State>  {
+  constructor(props: Props): void {
+    super(props);
+    this.state = {
+      errorMessages: []
+    };
+  }
+  
+  componentWillReceiveProps(nextProps: Props): void {
+    if(nextProps.formState && nextProps.validations) {
+      var failedValidations = _.filter(nextProps.validations, validation => !validation.checkFunc(nextProps.formState));
```

Comment:
Is there a reason lodash is being used here instead of the builtin function?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

Reply:
Yes, my brain was in lodash mode at the time.  Corrected!